build.sbt

```diff
@@ -373,8 +375,23 @@ lazy val libManagedSettings = Seq(
       }
     cachedFun(filesToWatch).toSeq
   },
-  Compile / sourceGenerators += (Compile / genProps).taskValue,
-  Compile / resourceGenerators += (Compile / genSchemas).taskValue,
+  Compile / genVersion := {
+    val resourceDir = (Compile / resourceManaged).value
+    val outFile = resourceDir / "org" / "apache" / "daffodil" / "lib" / "VERSION"
+    if (!outFile.exists || IO.read(outFile) != version.value) {
```

Comment:
I'm surprised you have to write this logic. Shouldn't sbt already have this kind of logic built in somewhere? 

I mean if you overwrite a file with the exact same contents, sbt doesn't rebuild based on that normally does it? It's not purely date-based reasoning. I thought it created a hash value and only rebuilds if both the date and hash value changed.

Reply:
After a little testing and a little digging in the sbt code (I can't find much documentation about this), it looks like sbt uses hashes for things in `src/main/scala/` and uses last modification time for `src/main/resources`. I guess he concern is resources could be very large and could slow things down? I think it also uses last modification time for both managed sources and resources. And since this is a managed resource it uses last modification time.

I do think it's configurable, but I'd rather not change it--the more settings we tweak in sbt the more complicated it gets, and it's already too complicated. And for sbt source/resourceGenerators it's common practice to do something like this so we only generate files when needed.